mmgen/models/architectures/stylegan/generator_discriminator_v3.py

```diff
@@ -0,0 +1,236 @@
+            evaluation, we use 'single' style mode. `['mix', 'single']` are
+            currently supported. Defaults to 'mix'.
+        eval_style_mode (str, optional): The evaluation mode of style mixing.
+            Defaults to 'single'.
+        mix_prob (float, optional): Mixing probability. The value should be
+            in range of [0, 1]. Defaults to 0.9.
+        rgb2bgr (bool, optional): Whether to reformat the output channels
+                with order `bgr`. We provide several pre-trained StyleGAN3
+                weights whose output channels order is `rgb`. You can set
+                this argument to True to use the weights.
+        pretrained (str | dict, optional): Path for the pretrained model or
+            dict containing information for pretained models whose necessary
+            key is 'ckpt_path'. Besides, you can also provide 'prefix' to load
+            the generator part from the whole state dict. Defaults to None.
+        synthesis_cfg (dict, optional): Config for synthesis network. Defaults
+            to dict(type='SynthesisNetwork').
+        mapping_cfg (dict, optional): Config for mapping network. Defaults to
+            dict(type='MappingNetwork').
+    """
+
+    def __init__(self,
+                 out_size,
+                 style_channels,
+                 img_channels,
+                 noise_size=512,
+                 c_dim=0,
+                 default_style_mode='mix',
+                 eval_style_mode='single',
+                 mix_prob=0.9,
+                 rgb2bgr=False,
+                 pretrained=None,
+                 synthesis_cfg=dict(type='SynthesisNetwork'),
+                 mapping_cfg=dict(type='MappingNetwork')):
+        super().__init__()
+        self.noise_size = noise_size
+        self.c_dim = c_dim
+        self.style_channels = style_channels
+        self.out_size = out_size
+        self.img_channels = img_channels
+        self.rgb2bgr = rgb2bgr
+        self._default_style_mode = default_style_mode
+        self.default_style_mode = default_style_mode
+        self.eval_style_mode = eval_style_mode
+
+        self._synthesis_cfg = deepcopy(synthesis_cfg)
+        self._synthesis_cfg.setdefault('style_channels', style_channels)
+        self._synthesis_cfg.setdefault('out_size', out_size)
+        self._synthesis_cfg.setdefault('img_channels', img_channels)
+        self.synthesis = build_module(self._synthesis_cfg)
+
+        self.num_ws = self.synthesis.num_ws
+        self._mapping_cfg = deepcopy(mapping_cfg)
+        self._mapping_cfg.setdefault('noise_size', noise_size)
+        self._mapping_cfg.setdefault('c_dim', c_dim)
+        self._mapping_cfg.setdefault('style_channels', style_channels)
+        self._mapping_cfg.setdefault('num_ws', self.num_ws)
+        self.style_mapping = build_module(self._mapping_cfg)
+
+        if pretrained is not None:
+            self._load_pretrained_model(**pretrained)
```

Comment:
`self.init_weights` is missing.

Reply:
Well, weights initialization is done separately in each module (mostly following the default initialization in `PyTorch`). StyleGAN2 also omit this function. If I found differences during training stylegan3, an extra PR will be submit.